cmds/signin/signin.go

```diff
@@ -141,3 +149,23 @@ func cmdSignin(cmd *cobra.Command, _ []string) error {
 
 	return nil
 }
+
+// Return an appropriate exit code based on whether we have credentials.
+// Useful for shell scripting around 'taskcluster signin' calls.
+func checkSignin() error {
+	var creds *tcclient.Credentials
+	if config.Credentials != nil {
+		creds = config.Credentials.ToClientCredentials()
+	}
+	auth := auth.New(creds)
+	result, err := auth.CurrentScopes()
+	if err != nil {
+		// Don't want an os.Exit() in case it causes scripting loops when used.
+		return fmt.Errorf("failed to check credentials: %s", err)
+	}
+	if len(result.Scopes) == 0 {
+		fmt.Println("No valid credentials were found")
```

Comment:
More accurate to say "No valid credentials were found, or credentials have no scopes".

I think this is a good way to check for access all the same, but if someone found themselves in the weird case of having a clientId with no scopes, the message might help them figure out what's going on.